network/common/src/test/java/org/apache/spark/network/buffer/LargeByteBufferHelperSuite.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.buffer;
+
+import java.io.*;
+import java.nio.channels.FileChannel;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class LargeByteBufferHelperSuite {
+
+  @Test
+  public void testMapFile() throws IOException {
+    File testFile = File.createTempFile("large-byte-buffer-test", ".bin");
+    try {
+      testFile.deleteOnExit();
+      OutputStream out = new FileOutputStream(testFile);
+      byte[] buffer = new byte[1 << 16];
+      long len = ((long)buffer.length) + Integer.MAX_VALUE + 1;
```

Comment:
While it's always nice to have more test, this is what I worry about:

    $ time dd if=/dev/zero of=/tmp/2gb bs=1024 count=$((2*1024*1024))
    2097152+0 records in
    2097152+0 records out
    2147483648 bytes (2.1 GB) copied, 9.05902 s, 237 MB/s
    
    real    0m9.061s
    user    0m1.337s
    sys     0m4.270s

That that's with a fast local SSD. Most of the underlying logic here is the same as the memory-based test, isn't it? Feels like the only extra things being tested are:

- The ability of the jdk to mmap a region of a file larger than 2gb
- The ability of your code to wrap a large file into multiple byte buffers

The only test I see as really interesting is the second. Using a file > 2g would only really help test the first.